gpu_support/test_cuda.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+# Test CUDA
+cuda_install_dir="${EESSI_SOFTWARE_PATH/versions/host_injections}"
+if [ -d ${cuda_install_dir}/modules/all ]; then
+  module use ${cuda_install_dir}/modules/all/
+else
+  echo "Cannot test CUDA, modules path does not exist, exiting now..."
+  exit 1
+fi
+module load CUDA
+ret=$?
+if [ $ret -ne 0 ]; then
+  echo "Could not load CUDA even though modules path exists..."
+  exit 1
+fi
+tmp_dir=$(mktemp -d)
+cp -r $EBROOTCUDA/samples $tmp_dir
+current_dir=$PWD
+cd $tmp_dir/samples/1_Utilities/deviceQuery
+make HOST_COMPILER=$(which g++) -j
```

Comment:
`g++` here needs to be coming from either the compat layer or EESSI to ensure we are using the correct loader. Since we already load the CUDA module, we might as well load GCC above.